src/main/resources/static/api.yml

```diff
@@ -0,0 +1,616 @@
+paths:
+  /treatments:
+    get:
+      tags:
+        - Doctors
+      summary: Get all treatments
+      description: ''
+      operationId: getAllTreatments
+      responses:
+        '200':
+          description: All treatments successfully fetched
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Treatment'
+  /patients:
+    get:
+      tags:
+        - Doctors
+        - Nurses
+      summary: Get all patient files
+      description: ''
+      operationId: getAllPatients
+      responses:
+        '200':
+          description: Patient files successfully fetched
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Patient'
+    post:
+      tags:
+        - Doctors
+      summary: Add a patient
+      description: Adds provided patient to database
+      operationId: addPatient
+      requestBody:
+        description: Patient info to be added to the database
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/NewPatient'
+      responses:
+        '200':
+          description: Patient successfully added to database
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Patient'
+  /patients/{patientId}:
+    get:
+      tags:
+        - Doctors
+        - Nurses
+      summary: Get patient file with the given ID
+      description: ''
+      operationId: getPatient
+      parameters:
+        - name: patientId
+          in: path
+          description: Unique patient identifier from Patients table
+          required: true
+          schema:
+            type: string
+            example: UUID
+      responses:
+        '200':
+          description: Patient file successfully fetched
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Patient'
+    put:
+      tags:
+        - Doctors
+      summary: Edit patient file of the given ID
+      description: ''
+      operationId: editPatient
+      parameters:
+        - name: patientId
+          in: path
+          description: ID of the patient that will be edited
+          required: true
+          schema:
+            type: string
+            example: UUID
+      requestBody:
+        description: Patient's JSON file that will replace existing one
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/NewPatient'
+
+      responses:
+        '200':
+          description: patient successfully modified
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Patient'
+    delete:
+      tags:
+        - Doctors
```

Comment:
lets split into logical entities like patients. appointments and so on
because API is the thing to show to the **client** what they can do, not for the user. also if some endpoints are accessible for both roles, it leads to API duplication

Reply:
Makes sense, will apply that from now on regarding API design. The tags are based on enitiy/controller this commit. For the 'particular appointment for particular patient' endpoint: for prescriptions (formerly handled as appointments) there is the `/prescriptions/{prescriptionId}` endpoint witch fetches a particular prescription, no need to specify the patient since each prescription is only associated to one patient. For events, a  GET endpoint was added to `/events/{eventId}` this commit to fetch a particular event which again are associated to only one prescription. In either case, it is not necessary to specify the patient, just the prescription/event ID which can be obtained from the endpoints that return all prescriptions/events for the patient.